File: CX/ClientApp/src/servicios/ServicioEstadoAceptacion.js
import { ProcesarDatosApi } from "./ApiFetch";
const controlador = "estadoaceptacion";

export const ObtenerEstadoAceptacion = async () => {
    const url = `${controlador}/obtenerestadoaceptacion`;
    return await ProcesarDatosApi('GET', url);
}

export const InsertarEstadoAceptacion = async (data) => {
    const url = `${controlador}/insertarestadoaceptacion`;
    return await ProcesarDatosApi('POST', url, data);
}
export const ActualizarEstadoAceptacion = async (data) => {
    const url = `${controlador}/actualizarestadoaceptacion`;
    return await ProcesarDatosApi('PUT', url, data);
}
export const InactivarEstadoAceptacion = async (id) => {
    const url = `${controlador}/eliminarestadoaceptacion?idEstadoAceptacion=${id}`;
    return await ProcesarDatosApi('DELETE', url);
}

export const ObtenerEstadoAceptacionPorID = async (id) => {
    const url = `${controlador}/ObtenerEstadoAceptacionPorID/${id}`;
    return await ProcesarDatosApi('GET', url);
}

File: BussinesLogic/Servicios/ServicioEstadoAceptacion.cs
using DataAccess.StoredProcedures;
using Dominio.Dto;
using Dominio.Entiti;
using Dominio.Interfaces.Aplicacion.EstadoAceptacion;
using Dominio.Interfaces.Infraestructura.BaseDatos;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;

namespace BussinesLogic.Servicios
{
    public class ServicioEstadoAceptacion : IServicioEstadoAceptacion
    {
        private readonly IRepositorioEstadoAceptacion repositorioEstadoAceptacion;

        public ServicioEstadoAceptacion(IRepositorioEstadoAceptacion repositorioEstadoAceptacion)
        {
            this.repositorioEstadoAceptacion = repositorioEstadoAceptacion;
        }

        public async Task<List<DtoEstadoAceptacion>> ObtenerEstadoAceptacion()
        {
            return await this.repositorioEstadoAceptacion.ObtenerEstadoAceptacion();
        }

        public async Task<DtoRespuestaSP> InsertarEstadoAceptacion(EntitiEstadoAceptacion entitiEstadoAceptacion)
        {
            return await this.repositorioEstadoAceptacion.InsertarEstadoAceptacion(entitiEstadoAceptacion);
        }
        public async Task<DtoRespuestaSP> ActualizarEstadoAceptacion(EntitiEstadoAceptacion entitiEstadoAceptacion)
        {
            return await this.repositorioEstadoAceptacion.ActualizarEstadoAceptacion(entitiEstadoAceptacion);
        }
        public async Task<DtoRespuestaSP> EliminarEstadoAceptacion(int idEstadoAceptacion)
        {
            return await this.repositorioEstadoAceptacion.EliminarEstadoAceptacion(idEstadoAceptacion);
        }
        public async Task<DtoEstadoAceptacion> ObtenerEstadoAceptacionPorID(int idEstadoAceptacion)
        {
            return await this.repositorioEstadoAceptacion.ObtenerEstadoAceptacionPorID(idEstadoAceptacion);
        }
    }
}

